### PR TITLE
[misc] Avoid unnecessary linking when git commit hash not changed

### PR DESCRIPTION
**Pull request by yuanming-hu**
Related issue = #856 

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

### CODE REVIEW

File: misc/generate_commit_hash.py [hunk 1]
```diff
@@ -10,11 +10,12 @@
...
 # First read the file to see if an update is needed
 # This reduces unnecessary file changes/linkings
-with open(output_fn, 'r') as f:
-    old_content = "\n".join(f.readlines())
-    if old_content == content:
-        # No update needed
-        exit(0)
+if os.path.exists(output_fn):
+    with open(output_fn, 'r') as f:
+        old_content = "\n".join(f.readlines())
```
Comment (archibate) [inline misc/generate_commit_hash.py:15]: Please use `f.read()` instead :)

**Comment by archibate** [comment]
We may want something like `alias make='make generate_commit_hash && make libtaichi_core.so'`?

**Comment by yuanming-hu** [comment]
Thanks. I implemented the review. On my end, unnecessary linkings are gone now.

**Comment by archibate** [comment]
Working! Great! Thank for your hardwork! Now people can `alias mp='make && python'` and no worrying waste of time!

**Pull request merged**